### PR TITLE
Feature/use parameter validation L-Z

### PR DESCRIPTION
**Pull request by dgostencnik**
apart from adapting tests and processes to use function for parameter validation

- some new tests were added (linear_scale_range, lt, lte, min, multiply, neq)
- `sum` and `min` were changed a bit as error validation was missing
- some formatting related changes in py and js files

### CODE REVIEW

File: src/pg_to_evalscript/javascript_processes/normalized_difference.js [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 
-  if (x === undefined) {
-    throw new Error("Mandatory argument `x` is not defined.");
-  }
-
-  if (y === undefined) {
-    throw new Error("Mandatory argument `y` is not defined.");
-  }
+  validateParameter({
+    processName: "normalized_difference",
```
Comment (Ardweaden) [inline src/pg_to_evalscript/javascript_processes/normalized_difference.js:5]: These two are not `nullable`.

**Review by Ardweaden — CHANGES_REQUESTED** [review]

File: src/pg_to_evalscript/javascript_processes/last.js [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
 
-  if (data === null || data === undefined) {
-    throw new Error("Mandatory argument `data` is either null or not defined.");
-  }
+  validateParameter({
+    processName: "last",
+    parameterName: "data",
+    value: data,
+    required: true,
+    nullable: false,
```
[the remark this answers is not in the document]
Comment (dgostencnik) [reply src/pg_to_evalscript/javascript_processes/last.js:9]: was added

File: src/pg_to_evalscript/javascript_processes/max.js [hunk 3]
```diff
@@ -1,24 +1,30 @@
...
+    value: data,
+    required: true,
+    array: true,
+  });
+
+  validateParameter({
+    processName: "max",
+    parameterName: "ignore_nodata",
+    value: ignore_nodata,
+    allowedTypes: ["boolean"],
```
[the remark this answers is not in the document]
Comment (dgostencnik) [reply src/pg_to_evalscript/javascript_processes/max.js:17]: was changed in all occurrences of ignore_nodata

File: src/pg_to_evalscript/javascript_processes/neq.js [hunk 4]
```diff
@@ -2,13 +2,19 @@ function neq(arguments) {
...
+    value: x,
+    required: true,
+  });
+
+  validateParameter({
+    processName: "neq",
+    parameterName: "y",
+    value: y,
+    required: true,
+  });
```
[the remark this answers is not in the document]
Comment (dgostencnik) [reply src/pg_to_evalscript/javascript_processes/neq.js:17]: was added

**Pull request merged**